### PR TITLE
Feature dashboard page

### PR DESCRIPTION
**Pull request by zamiraskaff**
Description:

* Implements the User's Dashboard with a section for viewing parties and a button to discover movies (the button currently doesn't route anywhere)

Type of change:

[  ] Bug fix (non-breaking change which fixes an issue)

[ X ] New feature (non-breaking change which adds functionality)

[  ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

[  ] This change requires a documentation update

How Has This Been Tested?

* Tests for the users dashboard are passing

Checklist:

[ X] My code follows the style guidelines of this project

[ X] I have performed a self-review of my own code

[ X] My changes generate no new warnings

[ X] I have added tests that prove my fix is effective or that my feature works

[  ] New and existing unit tests pass locally with my changes
* New tests are passing, the old tests from the landing page are still failing since I was working from an old branch 

[ X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: spec/features/users/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @user1 = User.create!(name: 'Chad', email: 'chad1@gmail.com')
+    @user2 = User.create!(name: 'Jessica', email: 'jessica2@gmail.com')
+    @user3 = User.create!(name: 'Fiona', email: 'Fiona3@gmail.com')
+  end
+
+  it 'should display users name at the top of the page' do
+    visit user_path(@user1.id)
+
+    within '#header' do
+      expect(page).to have_content("Chad's Dashboard")
```
Comment (ajpulzone) [inline spec/features/users/show_spec.rb]: For more dynamic testing, we might want to consider using "#{@user1.name}'s Dashboard" in place of the hard coded "Chad's Dashboard".

Same in regards to Jessica's test below. (Nice sad path testing!)

File: spec/features/users/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe 'User Dashboard Page' do
+  before(:each) do
+    @user1 = User.create!(name: 'Chad', email: 'chad1@gmail.com')
+    @user2 = User.create!(name: 'Jessica', email: 'jessica2@gmail.com')
+    @user3 = User.create!(name: 'Fiona', email: 'Fiona3@gmail.com')
+  end
+
```
Comment (ajpulzone) [inline spec/features/users/show_spec.rb:9]: Can we add the following test to ensure that the home button and Viewing Party title show up on every page?
*If you know of a more efficient way to ensure something is showing up on every page, please let me know!"
*I added a global partial to render the Home button and Viewing Party title to every new page that is created on another branch, so the test should pass after we get everything merged (it's ok to put it in as failing right now)

it 'should contain title of Viewing Party and a Home button that will redirect the user
    the landing page' do
    visit root_path
    within('#dashboard') do
      expect(page).to have_content('Viewing Party')
      expect(page).to have_link('Home')
      click_link('Home')
      expect(current_path).to eq(root_path)
    end
  end

**Review by ajpulzone — CHANGES_REQUESTED** [review]
I have left some feedback regarding possible changes to be made prior to merge approval. If they make sense and you are comfortable/agree then please make and push up changes to branch. If not we can discuss tomorrow.

File: spec/features/users/show_spec.rb (hunk 1 above)
Comment (zamiraskaff) [reply spec/features/users/show_spec.rb]: I made the changes and added them to the pull request.

File: spec/features/users/show_spec.rb (hunk 2 above)
Comment (zamiraskaff) [reply spec/features/users/show_spec.rb:9]: I added the test, it is currently failing but we can resolve that when it's merged.

**Pull request merged**